### PR TITLE
LEAF-2034 Add File Tracking to Editors

### PR DESCRIPTION
**Pull request by jrodcode**
This feature enables users to make changes to different sections of the admin panel and track multiple versions of edited files. The system allows for seamless merging of edited files in case of content changes. This streamlines the editing process, prevents errors, and ensures that all changes are recorded and stored for future reference.

### CODE REVIEW

File: LEAF_Request_Portal/admin/index.php [hunk 1]
```diff
@@ -2,206 +2,167 @@
...
-
 error_reporting(E_ERROR);
-
 require_once '../globals.php';
 require_once LIB_PATH . '/loaders/Leaf_autoloader.php';
-
 header('X-UA-Compatible: IE=edge');
-
 $login->setBaseDir('../');
-
```
Comment (jampaul3) [inline LEAF_Request_Portal/admin/index.php:20]: Why did you remove every blank line in this file? Blank lines do help in readability

File: LEAF_Request_Portal/api/controllers/EmailTemplateController.php [hunk 2]
```diff
@@ -8,80 +8,66 @@
...
-            echo $verified;
-        } else {
-            $this->index['POST'] = new ControllerMap();
-
-            $this->index['POST']->register('emailTemplates/[text]', function ($args) use ($emailTemplate) {
-                return $emailTemplate->setEmailTemplate($args[0]);
-            });
-
-            return $this->index['POST']->runControl($act['key'], $act['args']);
-        }
```
Comment (jampaul3) [inline LEAF_Request_Portal/api/controllers/EmailTemplateController.php:66]: Is there a reason for undoing the RESTfulResponse changes that were released in the last push to prod? This verified check needs to be here in order to echo the response if needed.

File: LEAF_Request_Portal/api/controllers/EmailTemplateController.php [hunk 3]
```diff
@@ -8,80 +8,66 @@
...
-        } else {
-            $this->index['DELETE'] = new ControllerMap();
-
-            $this->index['DELETE']->register('emailTemplates/[text]', function ($args) use ($emailTemplate) {
-                return $emailTemplate->removeCustomEmailTemplate($args[0]);
-            });
-
-            return $this->index['DELETE']->runControl($act['key'], $act['args']);
-        }
+        $this->verifyAdminReferrer();
```
Comment (jampaul3) [inline LEAF_Request_Portal/api/controllers/EmailTemplateController.php:66]: Same here.

File: LEAF_Request_Portal/api/controllers/TemplateFileHistory.php [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+    public function post($act)
+    {
+        $db = $this->db;
+        $login = $this->login;
+        $templateFileHistory = $this->templateFileHistory;
+
+        $this->verifyAdminReferrer();
+
+        $this->index['POST'] = new ControllerMap();
```
Comment (jampaul3) [inline LEAF_Request_Portal/api/controllers/TemplateFileHistory.php:59]: This file will need the verified check in order to work properly.

File: LEAF_Request_Portal/api/controllers/TemplateReportsController.php [hunk 5]
```diff
@@ -8,83 +8,82 @@
...
-        $verified = $this->verifyAdminReferrer();
-
-        if ($verified) {
-            echo $verified;
-        } else {
-            $this->index['POST'] = new ControllerMap();
-
-            $this->index['POST']->register('reportTemplates', function () use ($templateReports) {
-                return $templateReports->newReportTemplate($_POST['filename']);
-            });
```
Comment (jampaul3) [inline LEAF_Request_Portal/api/controllers/TemplateReportsController.php:62]: Again this should remain.

File: LEAF_Request_Portal/api/index.php [hunk 6]
```diff
@@ -43,137 +36,171 @@
...
+});
+
+$controllerMap->register('templateEmailHistoryMergeFile', function () use ($db, $login, $action) {
+    $TemplateFileHistoryController = new Portal\TemplateFileHistoryController($db, $login);
+    $TemplateFileHistoryController->handler($action);
 });
 
 $controllerMap->register('reportTemplates', function () use ($db, $login, $action) {
     $TemplateReportsController = new Portal\TemplateReportsController($db, $login);
-    echo $TemplateReportsController->handler($action);
```
Comment (jampaul3) [inline LEAF_Request_Portal/api/index.php:176]: I believe all these handler lines need to have the echo on them.

**Review by jampaul3 — CHANGES_REQUESTED** [review]

**Review by maxrichardjr — APPROVED** [review]

File: libs/logger/formatters/LoggableTypes.php [hunk 7]
```diff
@@ -13,7 +13,6 @@ class LoggableTypes
     const INDICATOR = "indicator";
     const PORTAL_GROUP = "portal_group";
     const WORKFLOW = "workflow";
-    const WORKFLOW_NAME = "workflow_name";
```
Comment (pete-nerantzinis) [inline libs/logger/formatters/LoggableTypes.php:16]: Why is this being removed.

**Review by pete-nerantzinis — CHANGES_REQUESTED** [review]
